File: app/src/main/java/io/zenandroid/greenfield/Application.java
package io.zenandroid.greenfield;

import android.content.Context;

import io.zenandroid.greenfield.dagger.Injector;

public class Application extends android.app.Application {

	private static Application instance;

	static {
		//
		// Note: if using vector images, you may want to do uncomment this line
		//
//		AppCompatDelegate.setCompatVectorFromResourcesEnabled(true);
	}

	@Override
	protected void attachBaseContext(Context base) {
		super.attachBaseContext(base);
		Injector.init(this);
	}

	@Override
	public void onCreate() {
		super.onCreate();
		instance = this;
	}

	public static Application getInstance() {
		return instance;
	}
}

File: app/src/main/java/io/zenandroid/greenfield/Application.kt
package io.zenandroid.greenfield

import android.content.Context

import io.zenandroid.greenfield.dagger.Injector

class Application : android.app.Application() {

    override fun attachBaseContext(base: Context) {
        super.attachBaseContext(base)
        Injector.init(this)
    }

    override fun onCreate() {
        super.onCreate()
        instance = this
    }

    companion object {

        var instance: Application? = null
            private set

        init {
            //
            // Note: if using vector images, you may want to do uncomment this line
            //
            //		AppCompatDelegate.setCompatVectorFromResourcesEnabled(true);
        }
    }

}
